src/library/scala/collection/IterableOnce.scala

```diff
@@ -743,56 +765,91 @@ trait IterableOnceOps[+A, +CC[_], +C] extends Any { this: IterableOnce[A] =>
   @deprecated("Use foldRight instead of :\\", "2.13.0")
   @`inline` final def :\ [B](z: B)(op: (A, B) => B): B = foldRight[B](z)(op)
 
-  /** Folds the elements of this $coll using the specified associative binary operator.
+  /** Applies the given binary operator to the given initial value and all elements of this
+   *  $coll.
+   *
+   *  For each application of the operator, each operand is either an element of this
+   *  $coll, the initial value, or another such application of the operator.
+   *  $undefinedOrder $exactlyOnce The initial value may be used an arbitrary number of
+   *  times but at least once.
+   *
+   *  If this collection is ordered, then for any application of the operator, the
+   *  element(s) appearing in the left operand will precede those in the right.
+   *
+   *  $orderIndependentReduce In either case, it is also necessary that the initial value
+   *  be a neutral element for the operator, e.g. `Nil` for list concatenation or `1` for
+   *  multiplication.
+   *
    *  The default implementation in `IterableOnce` is equivalent to `foldLeft` but may be
    *  overridden for more efficient traversal orders.
    *
-   *  $undefinedorder
    *  $willNotTerminateInf
    *
-   *  @tparam A1     a type parameter for the binary operator, a supertype of `A`.
-   *  @param z       a neutral element for the fold operation; may be added to the result
-   *                 an arbitrary number of times, and must not change the result (e.g., `Nil` for list concatenation,
-   *                 0 for addition, or 1 for multiplication).
-   *  @param op      a binary operator that must be associative.
-   *  @return        the result of applying the fold operator `op` between all the elements and `z`, or `z` if this $coll is empty.
+   *  @tparam A1     The type parameter for the binary operator, a supertype of `A`.
+   *  @param z       An initial value; may be used an arbitrary number of times in the
+   *                 computation of the result; must be a neutral element for `op` for the
+   *                 result to always be the same across runs.
+   *  @param op      A binary operator; must be associative for the result to always be the
+   *                 same across runs.
+   *  @return        The result of applying `op` between all the elements and `z`, or `z`
+   *                 if this $coll is empty.
    */
   def fold[A1 >: A](z: A1)(op: (A1, A1) => A1): A1 = foldLeft(z)(op)
 
-  /** Reduces the elements of this $coll using the specified associative binary operator.
+  /** Applies the given binary operator to all elements of this $coll.
```

Comment:
It would be nice to use the name, as if to say, "reduction means...".
```
Reduces the elements of this $coll by applying the given operator to all elements of this $coll.
```
I don't mind the minor repetition.

Reply:
Yes, I see what you mean.

There are a few methods called `reduceSomething`. I remember from middle school that all dictionary definitions are circular. I wish I had an example handy that would make an adolescent laugh or despair.